### PR TITLE
Add password validation

### LINKED ISSUE #75: Add support for password validation to frontend/backend
Add support for password validation (length, complexity, etc) to the backend API as well as the frontend registration pages.

### PR DESCRIPTION
**Pull request by maurizi**
## Overview

This adds password validation to the frontend & backend. A password validation popup is shown when focused on the password input field.

### Demo
No password:
localhost_3003_register (9)

`mike`:
localhost_3003_register (10)

`mike12345`:
localhost_3003_register (11)

Password reset page:
localhost_3003_password-reset_TrSsgxpXvxPilDPi4X21A2ttlIE (1)

### Notes

 - This is _heavily_ cribbed from the client-side password validation implementation in [Temperate](https://github.com/azavea/temperate/pull/964) which is itself designed to match the default password rules of Django. There is no similar need to match Django's rules here, but they seemed as valid as any.
 - I first attempted to add the vendored `difflib.js` file and add a `.d.ts` declaration for the 1 method I'm calling (`quickRatio`). I ran into trouble with this and eventually settled for converting it to an unchecked TS file using an `@ts-nocheck` comment. :sob: 

## Testing Instructions

- `scripts/update`
 - Register a new user
 - Reset their password

Closes #75

### CODE REVIEW

**Comment by maurizi** [comment]
Updated the popup styling / copy per feedback from @tgilcrest 
localhost_3003_register (12)

File: src/client/components/Field.tsx [hunk 1]
```diff
@@ -1,7 +1,19 @@
 /** @jsx jsx */
 import React, { RefAttributes } from "react";
-import { Box, Input, InputProps, jsx, Label, Select, SelectProps, Styled } from "theme-ui";
+import {
+  Box,
+  Flex,
+  Heading,
```
Comment (pcaisse) [inline src/client/components/Field.tsx:6]: I see warning for this line in the client logs:

> client_1    | ./src/client/components/Field.tsx
client_1    |   Line 6:3:  'Heading' is defined but never used  @typescript-eslint/no-unused-vars

Maybe warnings should break the build?

File: src/shared/password-validator/difflib.ts [hunk 2]
```diff
@@ -0,0 +1,1692 @@
+// @ts-nocheck
```
[the remark this answers is not in the document]
Comment (maurizi) [reply src/shared/password-validator/difflib.ts:1]: I tried and failed to do this by doing exactly what you suggested - would be very happy to clean this up if you know how.

File: src/shared/password-validator/common-passwords.ts [hunk 3]
```diff
@@ -0,0 +1,1003 @@
+// From https://github.com/django/django/blob/master/django/contrib/auth/common-passwords.txt.gz
```
[the remark this answers is not in the document]
Comment (maurizi) [reply src/shared/password-validator/common-passwords.ts:1]: Yes, I'm not happy with it. Also not sure what the solution is - we arrived at a similar place in Temperate despite not liking doing it there either. That these are apparently quite popular passwords is uh.... elucidating.

FWIW Django has this checked into their repo as a `.tgz` file and uncompresses it at runtime, but I don't think that makes sense for our use-case.

File: src/shared/password-validator/difflib.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (maurizi) [reply src/shared/password-validator/difflib.ts:1]: I don't actually need to use `difflib` specifically at all, any library for checking word similarity would do, but as you note due to #179 I'd have to vendor whatever I used instead and probably run into similar issues declaring typings (which I think were also due to similar root issues to #179).

[the remark this answers is not in the document]
Comment (maurizi) [reply src/shared/password-validator/difflib.ts:1]: I added an issue to track replacing this vendored library https://github.com/PublicMapping/districtbuilder/issues/232

I asked on Slack about this and the consensus was to defer fixing this for now

**Pull request merged**